File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    sl, ss = 0, 0
    answer = 0
    
    for s in sizes:
        if s[0] > s[1]:
            one, two = s[0], s[1]
        else:
            one, two = s[1], s[0]
        sl = max(one, sl)
        ss = max(two, ss)
        if answer < sl * ss:
            answer = sl * ss
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int sl = 0, ss = 0;
        int one, two;
        
        for (int[] s : sizes) {
            if (s[0] > s[1]) {
                one = s[0];
                two = s[1];
            }
            else {
                one = s[1];
                two = s[0];
            }
            
            sl = Math.max(one, sl);
            ss = Math.max(two, ss);
            
            if (answer < sl * ss) answer = sl * ss;
        }
        return answer;
    }
}